tests/runner/testresource.go

```diff
@@ -0,0 +1,101 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package runner
+
+import (
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Disposable is an interface representing the disposable test resources
+type Disposable interface {
+	Name() string
+	Init() error
+	Dispose() error
+}
+
+// TestResources holds initial resources and active resources
+type TestResources struct {
+	resources           []Disposable
+	resourcesLock       sync.Mutex
+	activeResources     []Disposable
+	activeResourcesLock sync.Mutex
+}
+
+// Add adds Disposable resource to resources queue
+func (r *TestResources) Add(dr Disposable) {
+	r.resourcesLock.Lock()
+	defer r.resourcesLock.Unlock()
+	r.resources = append(r.resources, dr)
+}
+
+// dequeueResource dequeus Disposable resource from resources queue
+func (r *TestResources) dequeueResource() Disposable {
+	r.resourcesLock.Lock()
+	defer r.resourcesLock.Unlock()
+	if len(r.resources) == 0 {
+		return nil
+	}
+	dr := r.resources[0]
+	r.resources = r.resources[1:]
+	return dr
+}
+
+// pushActiveResource pushes Disposable resource to ActiveResource stack
+func (r *TestResources) pushActiveResource(dr Disposable) {
```

Comment:
similarly - "addToActiveResources"

Reply:
I wanted to make method name more descriptive for active resource stack. I would like to keep it as is now. if we need to change the names of methods which are private, let's change in the separated pr.